src/apexguru/apex-guru-service.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as vscode from 'vscode';
+import * as fspromises from 'fs/promises';
+import { CoreExtensionService } from '../lib/core-extension-service';
+import { ApexGuruAuthResponse, ApexGuruInitialResponse } from '../types';
+import * as Constants from '../lib/constants';
+
+export async function isApexGuruEnabledInOrg(outputChannel: vscode.LogOutputChannel): Promise<boolean> {
+    try {
+		const connection = await CoreExtensionService.getConnection();
+		const response:ApexGuruAuthResponse = await connection.request({
+			method: 'GET',
+			url: Constants.APEX_GURU_AUTH_ENDPOINT,
+			body: ''
+		});
+		return response.status == 'Success';
+	} catch(e) {
+		// This could throw an error for a variety of reasons. The API endpoint has not been deployed to the instance, org has no perms, timeouts etc,.
+		// In all of these scenarios, we return false.
+		const errMsg = e instanceof Error ? e.message : e as string;
+		outputChannel.error('***ApexGuru perm check failed with error:***' + errMsg);
+		outputChannel.show();
+		return false;
+	}
+}
+
+export async function runApexGuruOnFile(selection: vscode.Uri, outputChannel: vscode.LogOutputChannel) {
+	try {
+        const requestId = await initiateApexGuruRequest(selection, outputChannel);
+		// TODO: Logging the request Id for easy QA. Future stories will use this requestId to poll and retrieve the Apex Guru report.
+        outputChannel.appendLine('***Apex Guru request Id:***' + requestId);
+    } catch (e) {
+        const errMsg = e instanceof Error ? e.message : e as string;
+        outputChannel.appendLine('***Apex Guru initiate request failed***');
+        outputChannel.appendLine(errMsg);
+    }
+}
+
+export async function initiateApexGuruRequest(selection: vscode.Uri, outputChannel: vscode.LogOutputChannel) {
+	const fileContent = await fileSystem.readFile(selection.fsPath);
+	const base64EncodedContent = Buffer.from(fileContent).toString('base64');
+	const connection = await CoreExtensionService.getConnection();
+	const response: ApexGuruInitialResponse = await connection.request({
+		method: 'POST',
+		url: Constants.APEX_GURU_REQUEST,
+		body: JSON.stringify({
+			classContent: base64EncodedContent
+		})
+	});
```

Comment:
It looks like you are assuming that the response will always be of type ApexGuruInitialResponse. But is it possible that you do not get a response back that has a status field?
Besides the fact that this request method is template, the other reason why you aren't getting a compiler error here from typescript is because you put this type in a d.ts file which is unchecked when compiled.

I suggest you move your ApexGuruInitialResponse and other types into typescript officially (in a .ts - not a .d.ts file) to be compiled, then this will complain unless you cast with "as ApexGuruInitialResponse" to the request output. 

But do we really want to do this casting? I would guess that you would want to first check the status code of the response. But now I see that the request method is templated so it is inferring the output type.

Do we know if the response object universally as something like a response code that can be used? Typically when making http requests, we validate the response to see if it is a 2** or a 4** or a 5** before we try to parse the response body.

Reply:
I moved the ApexGuruInitialResponse out out of the d.ts file. I noticed we don't get 2**/4** responses as part of the response. I'll bring this up with the ApexGuru team tomorrow in the meeting. I hope it should be easy fix for them. Btw, the connection request also does not give the status codes. So, I'll have to leave this as is for now.